bin/nrnivmodl.in

```diff
@@ -166,138 +161,98 @@ if [ ! -d $MODSUBDIR ] ; then
   mkdir $MODSUBDIR
 fi
 
-files=`echo "$files" | sed 's/\.mod//g'`
+files=$(echo "$files" | sed 's/\.mod//g')
 
-if test "$files" = '*' ; then
+if [ "$files" = '*' ] ; then
   files=""
 fi
 
 cd $MODSUBDIR
-mdir="`pwd`"
-
-  # construct file to be included by makefile to specify mod to c rule when
-  # executed in $MODSUBDIR (child folder of launch location folder)
-  MODMAKE=makemod2c_inc
-  echo "" > $MODMAKE
-  for i in $files ; do
-    case "$i" in
-      /*) f=$i;; # absolute, fine as is
-      *)  f=../$i;; # relative
-    esac
-    base_name="`basename $f`"
-    dir_name="`dirname $f`"
-    # Note: indentation for shell lines in make rules must be a tab
-    echo "\
-./$base_name.c: `escape_spaces "$f.mod"`
+mdir="$PWD"
+
+# construct file to be included by makefile to specify mod to c rule when
+# executed in $MODSUBDIR (child folder of launch location folder)
+MODMAKE=makemod2c_inc
+touch "$MODMAKE"
+for i in $files ; do
+  case "$i" in
+    /*) f=$i;; # absolute, fine as is
+    *)  f=../$i;; # relative
+  esac
+  base_name="$(basename $f)"
+  dir_name="$(dirname $f)"
+  # Note: indentation for shell lines in make rules must be a tab
+  echo "\
+./$base_name.c: $(escape_spaces "$f.mod")
 	@printf \" -> \$(C_GREEN)NMODL\$(C_RESET) \$<\\\n\"
-	(cd \"`unhide_spaces $dir_name`\"; MODLUNIT=\$(NRNUNITS) \$(NOCMODL) $base_name.mod -o \"$mdir\")
+	(cd \"$(unhide_spaces "$dir_name")\"; MODLUNIT=\$(NRNUNITS) \$(NOCMODL) $base_name.mod -o \"$mdir\")
 
 ./$base_name.o: ./$base_name.c
 	@printf \" -> \$(C_GREEN)Compiling\$(C_RESET) \$<\\\n\"
-	\$(COMPILE) -I\"`unhide_spaces $dir_name`\" \$(INCLUDES) @CMAKE_CXX_COMPILE_OPTIONS_PIC@ -c \$< -o \$@
-" >> $MODMAKE
-  done
-  if test "$nrnivmodl_cfiles" != "" ; then
```

Comment:
@nrnhines I need your review about that, because it seems no one today set or use `nrnivmodl_cfiles`.

Reply:
But what is worth mentioning is that the all the logic for `hiding / unhiding spaces` was done for #850 i.e.
* user is working inside a directory path that contains spaces e.g. `/drive/Shared with me`
* user now performs `nrnivmodl mod-dir`

I think we can add such test in the neuron (e.g. inside CMake or just under test sub-directory ?) and make sure this feature is not broken. Previous discussion about this issue is in https://github.com/neuronsimulator/nrn/issues/846#issuecomment-736874242.